### PR TITLE
Fix issue related to CLIC pointer fetch while MRET in WB

### PR DESCRIPTION
**Pull request by silabs-oivind**
Signed-off-by: Oivind Ekelund <oivind.ekelund@silabs.com>

### CODE REVIEW

**Comment by Silabs-ArjanB** [comment]
@silabs-oysteink, @silabs-oivind  Where can I find a description of the problem fixed with this PR?

**Comment by silabs-oivind** [comment]
> @silabs-oysteink, @silabs-oivind Where can I find a description of the problem fixed with this PR?
@Silabs-ArjanB 
The problem here was that an mret in WB would not result in csr_restore_mret being asserted if the FSM was in the PONTER_FETCH state.
This in turn would result in mstatus, mintstatus and privilege level (on e40s) not being updated correctly upon mret.

**Comment by Silabs-ArjanB** [comment]
The whole POINTER_FETCH state has become a bit convoluted now that mret is also being handled there. We really should have an assertion now that none of the other synchronous events that we check for in the FUNCTIONAL state can possibly be true while in the POINTER_FETCH state (to be sure that we are not missing handling of other things while waiting in POINTER_FETCH)

For example !(exception_in_wb && exception_allowed) && !wfi_in_wb && !fencei_in_wb etc.

Basically we want similar assertions for all states other than FUNCTIONAL (so also e.g. for SLEEP).
